test/splitter.js

```diff
@@ -0,0 +1,160 @@
+const truffleAssert = require('truffle-assertions');
+const Splitter = artifacts.require("./Splitter.sol");
+
+contract('Splitter', async accounts => {
+
+    let gasCost = async receipt => {
+        const gasUsed = receipt.receipt.gasUsed;
+        const tx = await web3.eth.getTransaction(receipt.tx);
+        const gasPrice = tx.gasPrice;
+
+        return web3.utils.toBN(gasUsed * gasPrice);
+    };
+
+    let checkEventNotEmitted = async eventName => {
+        let result = await truffleAssert.createTransactionResult(splitter, splitter.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result,
+            eventName
+        );
+    }
+
+    const senderAddress = accounts[1];
+    const RecipientOneAddress = accounts[2];
+    const RecipientTwoAddress = accounts[3];
+    let splitter;
+
+    beforeEach("deploy and prepare", async function() {
+        splitter = await Splitter.new();
+    });
+
+    it('Sender sends 5 wei, 4 wei is split equally between recipients and 1 wei sent to the senders balance', async () => {
+
+        // Take snapshot of initial balances
+        let initSenderEthBalance = await web3.eth.getBalance(senderAddress);
+        initSenderEthBalance = web3.utils.toBN(initSenderEthBalance);
+
+        const initSenderContractBalance = await splitter.balances(senderAddress);
+        const initRecipientOneContractBalance = await splitter.balances(RecipientOneAddress);
+        const initRecipientTwoContractBalance = await splitter.balances(RecipientTwoAddress);
```

Comment:
Can you think something shorted than `ContractBalance`? My take: `Owed`.

Reply:
Also, are these not `0` really? And if they are not `0`, then it is an error. Perhaps a separate `it` test is warranted.

Reply:
> Also, are these not `0` really? And if they are not `0`, then it is an error. Perhaps a separate `it` test is warranted.

I was thinking the balance maybe above 0 because the data in storage has changed from running the test multiple times.

Would it be better to shutdown the ganache instance and restart it taking a snapshot of the initial, clean blockchain and then reinstate the original snapshot each time the test is run so it is always starts with all balances at 0?

Reply:
> > Also, are these not `0` really? And if they are not `0`, then it is an error. Perhaps a separate `it` test is warranted.
> 
> I was thinking the balance maybe above 0 because the data in storage has changed from running the test multiple times.
> 
> Would it be better to shutdown the ganache instance and restart it taking a snapshot of the initial, clean blockchain and then reinstate the original snapshot each time the test is run so it is always starts with all balances at 0?

Sorry, it just occurred to me that beforeEach() creates a new Splitter every time so the balance will always be 0. I will rectify that.